src/Altinn.App.Core/Helpers/Serialization/ModelSerializationService.cs

```diff
@@ -167,6 +183,24 @@ DataType dataType
             };
         }
 
+        // Validate the deserialized model to catch missing required fields early
```

Comment:
Kan godt hende vi bør gjøre dette, men jeg ville foretrekke at vi gikk gjennom `IValidator` og gikk (re)innførte validering av enkelte data element i stedet for å kopiere deler av `DataAnnotationValidator` her. Så vidt jeg husker er det litt mer komplisert om du ønsker å sjekke [Required] på egenskaper dypere i modell hierarkiet https://github.com/Altinn/app-lib-dotnet/blob/f7b91e0a6627169a56ad63823557d2a137640d96/src/Altinn.App.Core/Features/Validation/Default/DataAnnotationValidator.cs#L62-L78

Reply:
Selv om vi vanligvis ignorerer xsd fila (etter konvertering til json schema) så er det vel en case for at vi kan bruke xsd til å validere xml data ved opplasting for å sikre at opplastingen er gyldig (men ved utfylling i nettleser så laster vi ufullstendige innsendinger for mellomlagring mange ganger før man sender inn, og det er ganske dumt om dette brekker muligheten for å mellomlagre halvferdig utfyllinger slik vår frontend gjør det)